src/main/kotlin/com/jetbrains/snakecharm/lang/parser/SnakemakeLexer.kt

```diff
@@ -271,7 +281,9 @@ class SnakemakeLexer : PythonIndentingLexer() {
         val possibleKeywordPosition = currentPosition
         val possibleToplevelSectionKeyword = tokenText
 
-        if (possibleToplevelSectionKeyword !in KEYWORDS || tokenStart != myCurrentNewlineOffset) {
+        if (possibleToplevelSectionKeyword !in KEYWORDS && possibleToplevelSectionKeyword !in TOPLEVEL_KEYWORDS
```

Comment:
You have 3 usages in code of KEYWORDS an TOPLEVEL_KEYWORDS in the same conditions e.g. `tokenText in KEYWORDS || tokenText in TOPLEVEL_KEYWORDS` or `!(tokenText in KEYWORDS || tokenText in TOPLEVEL_KEYWORDS)`. Seems this 2 sets assumed to be used together and is better to make a constant set that merges them, otherwise next time is easy to miss one of these variables in simple condition. 

Also variables naming is confusing. `localrules` and `rule` are also toplevel keywords, but they are in set `KEYWORDS`. Here should be smth like `SnakemakeLexer.KEYWORDS_2_TOKEN_TYPE` and `SnakemakeAPI.WORKFLOW_TOPLEVEL_SECTION_KEYWORDS`

Reply:
https://github.com/JetBrains-Research/snakecharm/pull/405/commits/6cd9fef60e3547d2700fde5511e30895c37e1928